Add exception logging to FrontendDependencies visitor

In some cases the class path scaning may fail if there is a broken class on the classpath that gets referenced.
This fail with an undescriptive error like:
```
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 15 out of bounds for length 12
    at net.bytebuddy.jar.asm.ClassReader.readLabel (ClassReader.java:2679)
    at net.bytebuddy.jar.asm.ClassReader.createLabel (ClassReader.java:2695)
    at net.bytebuddy.jar.asm.ClassReader.readTypeAnnotations (ClassReader.java:2760)
    at net.bytebuddy.jar.asm.ClassReader.readCode (ClassReader.java:1937)
    at net.bytebuddy.jar.asm.ClassReader.readMethod (ClassReader.java:1514)
    at net.bytebuddy.jar.asm.ClassReader.accept (ClassReader.java:744)
    at net.bytebuddy.jar.asm.ClassReader.accept (ClassReader.java:424)
    at com.vaadin.flow.server.frontend.scanner.FrontendDependencies.visitClass (FrontendDependencies.java:494)
```
There should be at least an error log on what class was the one that the ClassReader failed on to help fixing/removing failing class.

--- a/flow-server/src/main/java/com/vaadin/flow/server/frontend/scanner/FrontendDependencies.java
+++ b/flow-server/src/main/java/com/vaadin/flow/server/frontend/scanner/FrontendDependencies.java
@@ -598,6 +598,11 @@ public class FrontendDependencies extends AbstractDependenciesScanner {
         try (InputStream is = url.openStream()) {
             ClassReader cr = new ClassReader(is);
             cr.accept(visitor, ClassReader.EXPAND_FRAMES);
+        } catch (Exception e) {
+            log().error(
+                    "Visiting class {} failed with {}.\nThis might be a broken class in the project.",
+                    className, e.getMessage());
+            throw e;
         }
 
         // all classes visited by the scanner, used for performance (#5933)

